release-tools/changelog.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3
+
+import argparse
+import datetime
+import os
+import re
+
+import debian.changelog
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="automatic changelog writer for snapd")
+    parser.add_argument("version", type=str, help="new snapd version")
+    parser.add_argument(
+        "changelog",
+        type=argparse.FileType("r"),
+        help="path to new changelog entry as generated by snappy-dch",
+    )
+    return parser.parse_args()
+
+
+debianish_distros = [
+    # first element is the name of the distro in the packaging directory
+    # second element is the debian distribution name
+    # third element is the suffix to add to the version number when generating
+    # the changelog
+    ("ubuntu-14.04", "trusty", "~14.04"),
+    ("ubuntu-16.04", "xenial", ""),
+    ("debian-sid", "unstable", "-1"),
+]
```

Comment:
Probably not warranted for a script like this, but an option is to make these tuples typed, so that you can refer to their components with something more readable than integer indexing. Something like:

```python3
from typing import NamedTuple

class Distro(NamedTuple):
    name: str  # name of the distro in the packaging directory
    debian_name: str  # debian distribution name
    version_suffix: str  # suffix to add to the version number in changelogs

debianish_distros = [
    Distro("ubuntu-14.04", "trusty", "~14.04"),
    Distro("ubuntu-16.04", "xenial", ""),
    Distro("debian-sid", "unstable", "-1"),
]

# ... then use them like ...

for distro in debianish_distros: # unchanged
    ...
    version=opts.version + distro.version_suffix # the payoff
```
Not offended if you just say "no thanks" to this. :-)

If memory serves, the ": str" type declarations are required for that method of declaring the namedtuple class. You could try removing them and see if I'm wrong, since they aren't used by anything.

Reply:
Nice, I like it, thanks for the suggestion. Applied